python/feathub/processors/local/type_utils.py

```diff
@@ -0,0 +1,49 @@
+#  Copyright 2022 The Feathub Authors
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+from typing import Dict, Type, cast
+
+import pandas as pd
+
+
+def cast_series_dtype(series: pd.Series, target_dtype: Type) -> pd.Series:
+    """
+    Cast the given Pandas Series to the given dtype.
+    """
+    if series.dtype == target_dtype:
+        return series
+
+    if target_dtype == str:
+        # Handle str type specially to avoid casting None value to 'None' str
+        return series.apply(lambda x: None if x is None else str(x))
+
+    return series.astype(target_dtype, errors="ignore")
```

Comment:
Can you explain what errors are ignored?

And if the goal is to keep the value as None if it is None, would it be simpler to do this:

```
return series.apply(lambda x: None if x is None else target_dtype(x))
```

Reply:
It ignores the error when casting a Series with nan value to integer. I updated the code to handle that case explicitly and the removed the `ignore`.